### ISSUE TITLE
Is there a way to avoid installing pre-releases?

### ISSUE THREAD

**Issue by sudoshindo** [ivan-hc/AppMan#30]
I found this utility today and wanted to try it out.
I installed balenaEtcher and the version I got was 1.19.4 which is a pre-release.
I was expecting it to download 1.18.11 instead.

I have an old code snippet that downloads Etcher and this is what it looks like:

```
curl -s https://api.github.com/repos/balena-io/etcher/releases/latest | grep "browser_download_url.*AppImage" | cut -d \" -f4 | xargs wget -q --show-progress -O balenaEtcher.AppImage
chmod 755 balenaEtcher.AppImage
```

Emphasis on the end of the URL `/releases/latest` which redirects to the latest one that IS NOT a pre-release.

**Comment by ivan-hc** [comment]
Hi, I set "releases" and not "releases/latest" as the default because many developers abandon the AppImage format or don't add it in some latest released release.

You can work around this by using the "--rollback" option, from there you will be shown a list of alternatives taken from the developer's main repository.

Uploading simplescreenrecorder-2024-02-07_22.28.28.mp4…

**Comment by ivan-hc** [comment]
EDIT: to prevent that the option "-u" updates everything including Etcher, you can use the option "lock" to disable the AM-updater

**Comment by ivan-hc** [comment]
About installing the latest release of a script, try this:
```
appman -d etcher
sed -i 's#releases -O -#releases/latest -O -#g' $(xdg-user-dir DESKTOP)/etcher
appman test appman test $(xdg-user-dir DESKTOP)/etcher
```
- downloads the script of etcher on your desktop;
- replaces "releases" with "seleases/latest";
-  and installs the script without patching the messages, in debug mode.

------------------------------------------------

About downgrading to the "latest" stable:
```
sed -i 's#releases -O -#releases/latest -O -#g' $HOME$(cat ~/.config/appman/appman-config)/etcher/AM-updater
appman -u etcher
```
it is enough to replace "releases" with "seleases/latest" in the AM-updater script of the app.

If you need it, I can create an option `--force-latest` that does it for you.

-------------------------------------------------
About authomatic configuration in custom distro...

Sure! You can use `echo`:
```
mkdir -p ~/.local/bin && echo 'export PATH=$PATH:$(xdg-user-dir USER)/.local/bin' >> ~/.bashrc && wget https://raw.githubusercontent.com/ivan-hc/AM/main/APP-MANAGER -O appman && chmod a+x ./appman && mv ./appman ~/.local/bin/appman
```
this is the command you see on the README to install AppMan in ~/.local/bin, then
```
echo "Applications/AppImages" | appman
appman -i foo bar xyz
```

**Comment by ivan-hc** [comment]
The new option `--force-latest` will be available in the next release
```
  '--force-latest')

	_no_sudo

	case $2 in
	'') 
		echo " USAGE: $AMCLI $1 [ARGUMENT]"; exit
		;;
	*)
		if ! test -d $APPSPATH/$2; then
			echo ' ERROR: "'$2'" is not installed, see "-f"'
		elif ! test -f $APPSPATH/$2/AM-updater; then
			echo ' ERROR: "'$AMCLI'" cannot manage updates for "'$2'"'
		elif ! grep -q "api.github.com" $APPSPATH/$2/AM-updater; then
			echo ' ERROR: "'$2'" source is not on Github'
		elif ! grep -q "releases -O -" $APPSPATH/$2/AM-updater; then
			echo ' ERROR: "'$2'" does not redirect to a generic "releases"'
		else
			sed -i 's#releases -O -#releases/latest -O -#g' $APPSPATH/$2/AM-updater
			$AMCLIPATH -u $2
		fi
		;;
	esac;;
```

**Comment by ivan-hc** [comment]
See https://github.com/ivan-hc/AM/releases/tag/5.6.4

https://private-user-images.githubusercontent.com/88724353/303220603-6083f43b-37be-4434-98c1-9ecf28d1b447.mp4?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDczNzAwMzAsIm5iZiI6MTcwNzM2OTczMCwicGF0aCI6Ii84ODcyNDM1My8zMDMyMjA2MDMtNjA4M2Y0M2ItMzdiZS00NDM0LTk4YzEtOWVjZjI4ZDFiNDQ3Lm1wND9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjA4VDA1MjIxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0NGM3Y2QyNzI3ZWFlNTMzMmRmNTIzYTI2MDBhYzk4NjUyMzg0NTQ2YjFkODA4ZGIyNDE2NjlkM2Y0MTUxMjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.i994yzDEFK8cZ1LspOiVv6fFk2i2ifVut4-PIwYqB2s

**Comment by sudoshindo** [comment]
Can I use `--force-latest` upon installation like this?

`appman install --force-latest etcher`

Or is it just

`appman --force-latest etcher`

**Comment by ivan-hc** [comment]
You should use it after you've installed the app.

None knows if the upstream developer has an Appimage or any build for Linux in "latest", as I said, many developers are discontinued in this practice, but to be sure that a "latest" build for Linux exists two are the ways:
1. go tho the repository;
2. install the app via AppMan/Am.

If I had enough money to have a server or any storage to download all the 1871 apps listed (also to speedup the download, against API's renstrictions of github) I had scripts to store all that packages, as I had a real repository of packages, not just scripts/PKGBUILDs.

AppImages are ment to be downloaded from the sites's developer, not to be distributed as packages from repositories like happesn with Pacman, APT and DNF. All I can do is to write scripts to check, find and download them.

Being "latest" a specific branch of "releases", we have not the absolute guarantee that the developer will still upload builds for Linux (may be them AppImages, TARs, ZIPs or any other format to archive a build for Linux).

**Comment by sudoshindo** [comment]
Okay, thank you for the info.

**Comment by ivan-hc** [comment]
> Okay, thank you for the info.

Don't worry, I'm doing everything I can to restore AppImage's dignity.

The other day my Debian Testing suggested removing `libfuse2`, an essential package for AppImages. If even a distribution like Debian which covers around fifteen different architectures and with releases lasting up to 7-8 years (see `oldoldstable`) advises you to remove this package... it's a bad sign. This is yet another sign that AppImages are close to being abandoned as a packaging format altogether.

My work here is amateur, but I hope that anyone who is better than me can give AppImage the right space, giving it back the support it deserves.

**Comment by ivan-hc** [comment]
@sudoshindo I've just updated the script, now you can do it:
```
appman -i --force-latest etcher
```
thank you for the idea

**Comment by sudoshindo** [comment]
Your work is not amateur.
Appman is already better than some other AppImage tools.

Other tools I used creates a new `.desktop` file for every version of a program and gives me duplicate menu entries. Appman does not do this and works just like other package managers. Appman is very convenient.

**Comment by ivan-hc** [comment]
Thank you, but...

> Your work is not amateur.

...it is amateur. I'm not a developer, I play with bash just for fun. Anyone can do better than me. All it takes is the will to do it, and few people want Appimages, including me.

It's just that one day I said to myself "why wait for someone to make a program for me? Maybe I can try to make one myself." And this is how AppMan and then AM were born... before I merged them into a single CLI.

I had an Avidemux AppImage in a subfolder, I didn't even know what an AppImage was, but I used it, unaware that there were more updated versions. Then I created some scripts locally, just to update to the latest version if available.

I saw that it worked, but since there was nothing similar around, I published it. 

AppMan is just a set of scripts that I have accumulated over 3 years of experiments, and being unemployed, I now have a lot of time to improve it (also because I have nothing better to do). That's all.
